### PR TITLE
add "JH_UniScheduler" batch_type

### PR DESCRIPTION
**Pull request by WangSimin123456**
JH UniScheduler was developed by JHINNO company. JH UniScheduler is a commercial software and uses "jsub" to submit tasks. Its overall architecture is similar to that of IBM's LSF. However, there are still some differences between them.

### Release Notes

- **New Features**
  - Introduced support for a new scheduler system, `JH_UniScheduler`, in the DPDispatcher Python package.
  - Added documentation and setup instructions for `JH_UniScheduler`.

- **Documentation**
  - Updated descriptions in the documentation to include `JH_UniScheduler`.
  - Added detailed setup requirements and usage instructions for `JH_UniScheduler`.

- **Testing**
  - Added new test cases to ensure proper functionality of `JH_UniScheduler`.
  - Included configuration files and simulation input commands for testing `JH_UniScheduler`.

- **Configuration**
  - Updated project configuration to include `jh_unischeduler` as a keyword.

### CODE REVIEW

File: doc/index.rst
Comment (njzjz) [inline doc/index.rst:1]: Please also add documentation to `doc/batch.md`.

File: dpdispatcher/machines/JH_UniScheduler.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+                        str,
+                        optional=True,
+                        default=None,
+                        doc=doc_custom_gpu_line,
+                    ),
+                ],
+                optional=False,
+                doc="Extra arguments.",
+            )
+        ]
```
[the remark this answers is not in the document]
Comment (njzjz) [reply dpdispatcher/machines/JH_UniScheduler.py:161]: This suggestion should be committed.

File: tests/devel_test_JH_UniScheduler.py [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+from dpdispatcher.machine import Machine
+from dpdispatcher.submission import Resources, Submission, Task
+
+sys.path.insert(0, os.path.abspath(os.path.join(os.path.dirname(__file__), "..")))
+
+# task_need_resources has no effect
+with open("jsons/machine_JH_UniScheduler.json") as f:
+    mdata = json.load(f)
+
```
[the remark this answers is not in the document]
Comment (njzjz) [reply tests/devel_test_JH_UniScheduler.py:13]: Did you actually run this file?

File: tests/test_JH_UniScheduler_script_generation.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+    setUpModule,  # noqa: F401
+)
+
+
+class TestJHUniSchedulerScriptGeneration(unittest.TestCase):
+    def setUp(self):
+        self.maxDiff = None
+
+    def test_shell_trival(self):
+        with open("jsons/machine_lazy_local_JH_UniScheduler.json") as f:
```
Comment (njzjz) [inline tests/test_JH_UniScheduler_script_generation.py:27]: It seems this file doesn't exist

**Review by njzjz — CHANGES_REQUESTED** [review]
Please fix all tests

**Comment by WangSimin123456** [comment]
> Please fix all tests

Thank you for your comment. It already says "All checks have passed". Can we merge this branch and close this issue? Or what should be done next?

**Comment by WangSimin123456** [comment]
> Please fix all tests

Thank you for your comment. It already says "All checks have passed". Can we merge this branch and close this issue? Or what should be done next?

**Review by njzjz — APPROVED** [review]

**Pull request merged**